06-find.md

```diff
@@ -454,8 +454,8 @@ about them."
 > find . -name '*.dat' | wc -l | sort -n
 > ~~~
 
-> ## Matching `ose.dat` but not `temp` {.challenge}
-> 
+> ## Matching `ose.dat` but not `temp` {}{.challenge}
```

Comment:
Normally Pandoc processes curly bracket as normal text:

~~~
$ pandoc <<EOF
{.foo}
EOF
<p>{.foo}</p>
~~~

But in some cases it uses curly bracket for HTML elements attributes. E.g.

    $ pandoc <<EOF
    ~~~ {.bar}
    foo
    ~~~
    EOF
    <pre class="bar"><code>foo</code></pre>
    $ pandoc <<EOF
    \`foo\`{.bar}
    EOF
    <p><code class="bar">foo</code></p>
    $ pandoc <<EOF
    # foo {.bar}  
    EOF
    <h1 id="foo" class="bar">foo</h1>


So far Pandoc allows you to do that for only a few elements (I think that only the three cases above). The trick part is when you have a heading ending with a code. Pandoc rules is that the curly bracket will refer to the near HTML element. So, in

~~~
> ## Matching `ose.dat` but not `temp` {}{.challenge}
~~~

the `{}` refers to `temp` and `{.challenge}` to the heading.

**Extra:** you can escape this behavior using

~~~
$ pandoc <<EOF
# foo \{.bar\}
EOF
<h1 id="foo-.bar">foo {.bar}</h1>
~~~

or

~~~
$ pandoc -f markdown-header_attributes <<EOF
# foo {.bar}
EOF
<h1 id="foo-.bar">foo {.bar}</h1>
~~~

> Whoa - why does this fix things? It looks like magic to me.

@gvwilson This makes sense now?